### ISSUE TITLE
Consider adding marking reason to tracer

### ISSUE THREAD

**Issue by mateoatr** [mono/linker#1874]
Currently the tracer only prints information about the source of the marking and the marked member, but drops any information about the `DependencyKind` (reason for marking). Printing this information might be useful for users of `--dump-dependencies`.

**Comment by sbomer** [comment]
Agreed, this would be nice, particularly with support from the analyzer. I had prototyped something that would include the DependencyKind, but IIRC I left it out because the tracer wouldn't support it.

**Comment by marek-safar** [comment]
I didn't that in #1228
